packages/plugin-vite/src/plugins/dev_server.ts

```diff
@@ -15,6 +15,22 @@ export function devServer(): Plugin[] {
       configureServer(server) {
         const IGNORE_URLS = /^\/(@(vite|fs|id)|\.vite)\//;
 
+        server.httpServer?.on("deno:request", (ev) => {
+          const { headers } = ev.req;
+          if (
+            headers.get("upgrade") !== "websocket" ||
+            // Vite uses websockets with protocol set to vite-hmr, vite-ping
+            headers.get("sec-websocket-protocol")?.startsWith("vite-")
+          ) {
```

Comment:
I wish every non-vite request was going through this route, as the current implementation looks too fragile

It is however only required and used for websocket for now.